irr/src/COpenGLSLMaterialRenderer.cpp

```diff
@@ -401,18 +401,32 @@ bool COpenGLSLMaterialRenderer::linkProgram()
 #endif
 
 		if (maxlen == 0) {
-			os::Printer::log("GLSL (> 2.x): failed to retrieve uniform information", ELL_ERROR);
-			return false;
+			// Work around driver bug: There are > 0 uniforms,
+			// yet the driver reports a max. uniform name length of 0.
+			if (GL.GetActiveUniformName) {
```

Comment:
AFAIK this is not standards compliant, function pointers can be any junk if version or extension requirements are not met.

Reply:
Reverted to the simpler workaround, this is too much of a hassle.